examples/charm++/allGather/Makefile

```diff
@@ -0,0 +1,20 @@
+CHARMC = ../../../bin/charmc $(OPTS)
+FLAGS = -g -Wall -Wno-unused-variable -Wno-sign-compare -Wno-class-memaccess -O2
```

Comment:
Can you make this Makefile consistent with others in Charm where the compiler flags can be passed into it rather than hardcoded here?